src/pages/Settings/TwoStepVerificationPage.vue

```diff
@@ -290,14 +260,57 @@ export default {
 							mfa_token: token,
 							id: mfaMethod.authId
 						}
-					});
-				})
+					})
 				.then(() => {
-					// ISSUE: This is not waiting for the mutaion to complete before firing.
-					// I thought the Return (line 288) above should be handling this, but isn't working
 					this.gatherMfaEnrollments();
-				});
+				})
+			});
 		},
+		formatMfaMethods(authEnrollments) {
+			console.log('formatMfaMethods(authEnrollments)', authEnrollments);
+
+			// When a user adds a text based recovery there are 2 new methods added to their authEnrollments
+			// 1) 1: {id: "sms|dev_tRkQzShZEIDAD0Q9", active: true, authenticator_type: "oob", oob_channel: "sms", name: "XXXXXXXX0367", …}
+			// 2) 2: {id: "voice|dev_tRkQzShZEIDAD0Q9", active: true, authenticator_type: "oob", oob_channel: "voice", name: "XXXXXXXX0367", …}
+			// Since the user only added 1 authentication method we only want to display 1 of these
+			// in their list of authentication methods
+
+			// My first thought was to remove one of the values based on the ID, since they come through with a similar ID
+			// ie. "sms|dev_tRkQzShZEIDAD0Q9" & "voice|dev_tRkQzShZEIDAD0Q9"
```

Comment:
you could split the ID on the `|` character then use the 2nd item in the resulting array.

How about for the the authenticator_type are there other `oob` options beyond voice/sms?

Also, I'm curious if we know which of these options the lender actually "selected" when setting this up? We should show the type they selected during setup.

If they add a phone number does it create both entries or is it only when they select "sms/text" recovery that 2 similar entries are created? If the duplicate is only created when selecting "sms/text" then we can always show that one if it's present.

Reply:
- Thanks for the .split("|") suggestion for the ID @mcstover, that's what I've implemented in my latest commit.
- I'm not aware of other oob options beyond voice/sms
- We don't have a way of knowing if a user chooses text or voice when setting up the MFA method.
- The duplicate method is shown when the user adds a phone number. During the recovery process I believe the user can choose to receive a text or voice call for recovery, leading to the duplicate entry in authEnrollments. For the front end it doesn't make sense to display both due to them being tied together (if one is deleted, the matching one is too.) That's why I wanted it filtered out for display.